### DISCUSSION TITLE
Content API in GitHub

### DISCUSSION THREAD

**Discussion by anvinraj** [community/community#24479] [Apps, API and Webhooks]
<p>Github content api is not giving the correct commit id in its ‘sha’ key value.</p>
<pre><code>https://github. ***.com/api/v3/repos/exampleowner-Management/examplerepo/contents/Compile/Teradata/Tables?access_token=*****
</code></pre>
<p>I’am using the above api and the sha key value is wrong.</p>
<p>the result of api call is :</p>
<pre><code>[
{
    "name": "file1.ddl",
    "path": "Compile/Tables/test",
    "sha": "1cef8efa8694678e3b7ab230a6a891afa1a1996d",
    "size": 8196,
    "url": "***",
    "html_url": "***",
    "git_url": "***",
    "download_url": "***",
    "type": "file",
    "_links": {
        "self": "***",
        "git": "***",
        "html": "***"
    }
}]
</code></pre>
<p>how to get the latest comimit for each file?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/anvinraj">@anvinraj</a>,</p>
<p>Thanks for being here! Are you getting the last event? ie:</p>
<ul>
<li>
</li>
</ul>
<p> <a href="https://developer.github.com/v3/activity/events/types/#pushevent" rel="nofollow noopener">the last push event</a>: Most recent commit done (on any branch), pushed by a user to the repo.</p>

**Comment by anvinraj** [comment]
<p>Hi <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>,</p>
<p>No, it is not showing the latest push details(commit id).</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/anvinraj">@anvinraj</a>,</p>
<p>For The Commits API, there’s an endpoint for fetching a list of commits on a repository. You could utilize the <code>sha</code> value to start listing commits from a specific SHA or branch:</p>
<p><a href="https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository" rel="nofollow noopener">https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository</a></p>
<p>Example:</p>
<p><a href="https://api.github.com/repos/github/linguist/commits?sha=dev" rel="nofollow noopener">https://api.github.com/repos/github/linguist/commits?sha=dev</a></p>
<p>If you’re interested in fetching every commit, I recommend leveraging pagination as well:</p>
<p><a href="https://developer.github.com/v3/#pagination" rel="nofollow noopener">https://developer.github.com/v3/#pagination</a></p>
<p>I hope this helps </p>
